homeassistant/components/wmspro/cover.py

```diff
@@ -75,3 +79,19 @@ async def async_stop_cover(self, **kwargs: Any) -> None:
             WMS_WebControl_pro_API_actionType.Stop,
         )
         await action()
+
+
+class WebControlProAwning(WebControlProCover):
+    """Representation of a WMS based awning."""
+
+    _attr_device_class = CoverDeviceClass.AWNING
+    _drive_action_desc = WMS_WebControl_pro_API_actionDescription.AwningDrive
+
+
+class WebControlProRollerShutter(WebControlProCover):
+    """Representation of a WMS based roller shutter or blind."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+    _drive_action_desc = (
+        WMS_WebControl_pro_API_actionDescription.RollerShutterBlindDrive
+    )
```

Comment:
Would it make sense to use either entity descriptions or to pass these as parameter to the class to allow for easy extension?

Reply:
Good point, but can you elaborate what "entity descriptions" are?

Reply:
Thanks. Yeah, venitian blinds (the next and 3rd cover to be supported) have additional behavior, because they can be tilted. But other than that they inherit the base cover open/close/position behavior.